fabric/src/test/java/TestStyledText.java

```diff
@@ -0,0 +1,413 @@
+
+        Assertions.assertEquals(
+                thirdExpected,
+                StyledText.fromComponent(thirdTestComponent).getString(PartStyle.StyleType.DEFAULT),
+                "StyledText.getString() returned an unexpected value.");
+
+        Assertions.assertEquals(
+                fourthExpected,
+                StyledText.fromComponent(fourthTestComponent).getString(PartStyle.StyleType.DEFAULT),
+                "StyledText.getString() returned an unexpected value.");
+    }
+
+    @Test
+    public void advancedComponentString_shouldProduceCorrectStyledText() {
+        final String testString = "§c§oitalicred§9§oblue§cnonitalic§oinherited§lbold§rafter";
+        final String[] expectedParts = {"§c§oitalicred", "§9§oblue", "§cnonitalic", "§oinherited", "§lbold", "§rafter"};
+
+        StyledText styledText = StyledText.fromString(testString);
+
+        for (String expectedPart : expectedParts) {
+            StyledTextPart partMatching = styledText.getPartMatching(Pattern.compile(expectedPart));
+
+            Assertions.assertNotNull(
+                    partMatching, "StyledText.getPartMatching() could not find a part matching: " + expectedPart);
+        }
+    }
+
+    @Test
+    public void normalizedStringCreation_shouldProduceCorrectString() {
+        final String badText = "ÀÀÀÀHello, ÀWorld!֎";
+        final String expected = "Hello, World!";
+
+        StyledText styledText = StyledText.fromString(badText);
+
+        Assertions.assertEquals(
+                expected,
+                styledText.getNormalized().getString(PartStyle.StyleType.NONE),
+                "StyledText.getNormalized().getString() returned an unexpected value.");
+    }
+
+    @Test
+    public void untrimmedText_shouldProduceTrimmedString() {
+        final String badText = "   Hello, World!  ";
+        final String expected = "Hello, World!";
+
+        StyledText styledText = StyledText.fromString(badText);
+
+        Assertions.assertEquals(
+                expected,
+                styledText.trim().getString(PartStyle.StyleType.NONE),
+                "StyledText.getString() returned an unexpected value.");
+    }
+
+    @Test
+    public void emptyText_shouldProduceEmptyString() {
+        StyledText styledText = StyledText.fromString("");
+
+        Assertions.assertTrue(styledText.isEmpty(), "StyledText.isEmpty() returned an unexpected value.");
+    }
+
+    @Test
+    public void blankText_shouldProduceBlankString() {
+        StyledText styledText = StyledText.fromString("    ");
+
+        Assertions.assertTrue(styledText.isBlank(), "StyledText.isBlank() returned an unexpected value.");
+    }
+
+    @Test
+    public void contains_shouldProduceCorrectResult() {
+        final String testString = "§c§oitalicred§9§oblue§cnonitalic§oinherited§lbold§rafter";
+        final String[] expectedParts = {"§c§oitalicred", "§9§oblue", "§cnonitalic", "§oinherited", "§lbold", "§rafter"};
```

Comment:
It would be interesting to test like `"icred§9§obl"`, and not just at part breaks.